internal/tenant/tenant.go

```diff
@@ -15,41 +16,43 @@ type Tenant struct {
 	authToken        string
 }
 
+// NewTenant creates a new instance of type Tenant
 func NewTenant(tenantServiceURL string, authToken string) Tenant {
 	return Tenant{
 		authToken:        authToken,
 		tenantServiceURL: tenantServiceURL,
 	}
 }
 
-func (t Tenant) GetTenantInfoByNamespace(api string, ns string) (TenantInfoList, error) {
+// GetTenantInfoByNamespace gets you InfoList of a tanent given a namespace and api url
```

Comment:
typo tanent == tenant 

I recommend you do a query replace over the whole source. I think you made this typo several times.